### DISCUSSION TITLE
Custom UI best practices

### DISCUSSION THREAD

**Discussion by vkhobor** [zitadel/zitadel#8853] [Q&A]
I was reading the docs about creating a custom UI, my aim is to implement it in flutter for mobile.
I have some questions. Some things weren't super clear to me.

1. To implement the custom UI do I need both a custom server(go) and client(flutter)?
   - reading this I don't see a clear separation https://zitadel.com/docs/assets/images/username-password-flow-3d9f45956742ccc19bbbb2d6538236ff.png
   - it seems to me the `v2/users/human` api should not be called from unsafe client software directly
  
2. How do I aquire a JWT token that I can send to the client?
   - The intention is that upon login I send back a JWT for the unsafe client to store locally, and send back on any resource request.
   - After that I can check the JWT for roles and userid on any resource path in my server, without calling out to Zitadel every time.
   
Thanks in advance

**Comment by zuffik** [comment]
@vkhobor were you able to resolve this issue? I'm facing this issue as well. It seems to me that the documentation does not mention this at all.

**Reply by vkhobor** [reply]
2. So what was clear after some digging is that JWT cannot be generated from the sessions api. 

 1. I believe yes these apis should be wrapped by your server, as they need authentication, they are not public

You can implement the full OIDC flow on your custom server, and then get a JWT as far as I know.

You should upvote the issue that is tracking "sessiontoken should be exchangable for jwt"

**Reply by vkhobor** [reply]
https://github.com/zitadel/zitadel/issues/7900#issue-2277264372
